templates/prometheusproxynetworkpolicy.go

```diff
@@ -0,0 +1,58 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package templates
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	netv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+var (
+	prometheusProxyPort     = intstr.FromInt(int(KubeRBACProxyPortNumber))
+	prometheusProxyProtocol = corev1.ProtocolTCP
+)
+
+var PrometheusProxyNetworkPolicyTemplate = netv1.NetworkPolicy{
+	Spec: netv1.NetworkPolicySpec{
+		Ingress: []netv1.NetworkPolicyIngressRule{
+			{
+				Ports: []netv1.NetworkPolicyPort{
+					{
+						Port:     &prometheusProxyPort,
+						Protocol: &prometheusProxyProtocol,
+					},
+				},
+			},
+		},
+		PolicyTypes: []netv1.PolicyType{
+			netv1.PolicyTypeIngress,
+		},
+		PodSelector: metav1.LabelSelector{
+			MatchExpressions: []metav1.LabelSelectorRequirement{
+				{
+					Key:      "prometheus",
+					Operator: metav1.LabelSelectorOpIn,
+					Values: []string{
+						"managed-ocs-prometheus",
+					},
```

Comment:
- shouldn't this be as below as I don't see a label `prometheus: managed-ocs-promethues` being set on prom pod, correct me if I'm wrong.
``` go
					Key:      "app.kubernetes.io/name",
					Operator: metav1.LabelSelectorOpIn,
					Values: []string{
						"prometheus",
					},
```

Reply:
I think prometheus operator sets this label `prometheus: managed-ocs-prometheus` on the prometheus-server pod(This is just a guess). But I checked Prometheus-server pod and it has the label `prometheus: managed-ocs-prometheus`  and `app: prometheus`. If you suggest using other one I'm fine with it, I just used `prometheus: managed-ocs-promethues` because it signifies that this is managed-ocs prometheus and not some other prometheus.

Reply:
ack, if the label exists on the prom pod then that's fine.